src/dex-ui/pages/dao/ProposalDetailsPage/ProposalDetailsPage.tsx

```diff
@@ -86,13 +94,25 @@ export function ProposalDetailsPage() {
       author,
       title,
       link,
+      isAdminApproved,
+      isContractUpgradeProposal,
+      data,
     } = proposalDetails;
 
     const isMultiSigProposal = daoType === DAOType.MultiSig;
     const isThresholdReached = approvalCount >= threshold;
     const memberCount = ownerIds.length;
+    const proxyId = (data as DAOUpgradeProposal)?.proxy ?? "";
+    const proxyAdmin = (data as DAOUpgradeProposal)?.proxyAdmin ?? "";
     /** TODO: Update contracts to support a "queued" status. */
-    const proposalStatus = status === ProposalStatus.Pending && isThresholdReached ? ProposalStatus.Queued : status;
+    const proposalStatus =
```

Comment:
I suggest moving this into a single re-usable utility function since it's used in two different components. Using `if` statements for the conditions would be more readable than a single nested ternary statement.

Reply:
Done